src/mono/netcore/System.Private.CoreLib/src/System/Environment.iOS.cs

```diff
@@ -0,0 +1,115 @@
+                case SpecialFolder.PrinterShortcuts:
+                case SpecialFolder.Windows:
+                case SpecialFolder.SystemX86:
+                case SpecialFolder.ProgramFilesX86:
+                case SpecialFolder.CommonProgramFilesX86:
+                case SpecialFolder.CommonDocuments:
+                case SpecialFolder.CommonAdminTools:
+                case SpecialFolder.AdminTools:
+                case SpecialFolder.CommonMusic:
+                case SpecialFolder.CommonPictures:
+                case SpecialFolder.CommonVideos:
+                case SpecialFolder.LocalizedResources:
+                case SpecialFolder.CommonOemLinks:
+                case SpecialFolder.CDBurning:
+                    return String.Empty;
+                
+                case SpecialFolder.Personal:
+                case SpecialFolder.LocalApplicationData:
+                    return NSDocumentDirectory;
+
+                case SpecialFolder.ApplicationData:
+                    // note: at first glance that looked like a good place to return NSLibraryDirectory 
+                    // but it would break isolated storage for existing applications
+                    return Path.Combine(NSDocumentDirectory, ".config");
+
+                case SpecialFolder.Resources:
+                    return NSLibraryDirectory; // older (8.2 and previous) would return String.Empty
+
+                case SpecialFolder.Desktop:
+                case SpecialFolder.DesktopDirectory:
+                    return Path.Combine(NSDocumentDirectory, "Desktop");
+
+                case SpecialFolder.MyMusic:
+                    return Path.Combine(NSDocumentDirectory, "Music");
+
+                case SpecialFolder.MyPictures:
+                    return Path.Combine(NSDocumentDirectory, "Pictures");
+
+                case SpecialFolder.Templates:
+                    return Path.Combine(NSDocumentDirectory, "Templates");
+
+                case SpecialFolder.MyVideos:
+                    return Path.Combine(NSDocumentDirectory, "Videos");
+
+                case SpecialFolder.CommonTemplates:
+                    return "/usr/share/templates";
+
+                case SpecialFolder.Fonts:
+                    return Path.Combine(NSDocumentDirectory, ".fonts");
+
+                case SpecialFolder.Favorites:
+                    return Path.Combine(NSLibraryDirectory, "Favorites");
+
+                case SpecialFolder.ProgramFiles:
+                    return "/Applications";
+
+                case SpecialFolder.InternetCache:
+                    return Path.Combine(NSLibraryDirectory, "Caches");
+
+                case SpecialFolder.UserProfile:
+                    return Environment.GetEnvironmentVariable("HOME");
+
+                case SpecialFolder.CommonApplicationData:
+                    return "/usr/share";
+
+                default:
+                    throw new ArgumentException($"Invalid SpecialFolder '{folder}'");
```

Comment:
In what situation will you get here?  Seems like you can instead just return string.Empty and delete the whole set of cases that do the same.

Reply:
Good point!